lib/time/time.go

```diff
@@ -0,0 +1,548 @@
+package time
+
+import (
+	"fmt"
+	"sort"
+	"time"
+
+	"go.starlark.net/starlark"
+	"go.starlark.net/starlarkstruct"
+	"go.starlark.net/syntax"
+)
+
+// ModuleName defines the expected name for this Module when used in the
+// starlark runtime.
+const ModuleName = "time"
+
+// Module time is a Starlark module of time-related functions.
+var Module = &starlarkstruct.Module{
+	Name: ModuleName,
+	Members: starlark.StringDict{
+		"duration":       starlark.NewBuiltin("duration", newDuration),
+		"parse_duration": starlark.NewBuiltin("parse_duration", parseDuration),
+		"parse_location": starlark.NewBuiltin("parse_location", parseLocation),
+		"now":            starlark.NewBuiltin("now", now),
+		"struct":         starlark.NewBuiltin("struct", starlarkstruct.Make),
+		"time":           starlark.NewBuiltin("time", newTime),
+		"parse_time":     starlark.NewBuiltin("time", parseTime),
+		"sleep":          starlark.NewBuiltin("sleep", sleep),
+		"from_timestamp": starlark.NewBuiltin("fromtimestamp", fromUnixTime),
+
+		"zero": Time{},
+
+		"nanosecond":  Duration(time.Nanosecond),
+		"microsecond": Duration(time.Microsecond),
+		"millisecond": Duration(time.Millisecond),
+		"second":      Duration(time.Second),
+		"minute":      Duration(time.Minute),
+		"hour":        Duration(time.Hour),
+	},
+}
+
+// LoadModule loads the time module.
+// It is concurrency-safe and idempotent.
+func LoadModule() (starlark.StringDict, error) {
+	return starlark.StringDict{
+		ModuleName: Module,
+	}, nil
+}
+
+// NowFunc is a function that generates the current time. Intentionally exported
+// so that it can be overridden, for example by applications that require their
+// Starlark scripts to be fully deterministic.
+var NowFunc = time.Now
+
+func parseDuration(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var d Duration
+	err := starlark.UnpackArgs("duration", args, kwargs, "d", &d)
+	return d, err
+}
+
+func parseLocation(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var s string
+	if err := starlark.UnpackArgs("location", args, kwargs, "s", &s); err != nil {
+		return nil, err
+	}
+	loc, err := time.LoadLocation(s)
+	if err != nil {
+		return nil, err
+	}
+
+	return starlark.String(loc.String()), nil
+}
```

Comment:
I don't get the use of this function... You put in a location string and get that string out? Why do you need this?

Reply:
this validates system clock time zone locations, users can use it to both confirm the timezone in question is supported by the OS. In the future This could be expanded to return a starlark "location" type that supports timezone conversion

Reply:
Can we then please name the function `validate_location()` or similar and return a bool!? Otherwise it is very confusing.

Reply:
Starlark's error-handling theory is not well developed. It doesn't have exceptions, and I don't think we really want to add them, but this does mean that callers need to "look before they leap", or need a mode in which they succeed but return an error value (e.g. None or perhaps a string) instead of failing.

Reply:
@alandonovan right, but `parseLocation` suggests that we somehow convert the given location to something else. Instead we simply return the same string in case of success or `nil` otherwise. Furthermore, the returned error is totally redundant with checking the string.... My suggestion is to simplify this to return a bool indicating if this is a valid location. So instead of 
```golang
l, err := parseLocation(..., loc)
if err != nil {...}
```
you can call
```golang
if isValidLocation(..., loc) {...}
```